src/hotspot/os/posix/perfMemory_posix.cpp

```diff
@@ -1196,29 +1194,19 @@ static void mmap_attach_shared(int vmid, char** addr, size_t* sizep, TRAPS) {
               "Process not found");
   }
 
+  // open the shared memory file for the give vmid
   char* filename = get_sharedmem_filename(dirname, vmid, nspid);
 
-  // copy heap memory to resource memory. the open_sharedmem_file
-  // method below need to use the filename, but could throw an
-  // exception. using a resource array prevents the leak that
-  // would otherwise occur.
-  char* rfilename = NEW_RESOURCE_ARRAY(char, strlen(filename) + 1);
-  strcpy(rfilename, filename);
+  // Handle the exception explicitly so we can free the strings.
```

Comment:
Suggestion:
// We don't use CHECK as we need to free the strings even if an exception occurred.

Reply:
Fixed.